R/get_recall_data.R

```diff
@@ -0,0 +1,158 @@
+#' }
+get_recall_make <- function(make, manufacturer = FALSE, limit = 25, partial = FALSE) {
+
+    # format the url string
+    if (manufacturer) {
+        url_ <- "https://vrdb-tc-apicast-production.api.canada.ca/eng/vehicle-recall-database/v1/recall/manufacturer-name/"
+    } else {
+        url_ <- "https://vrdb-tc-apicast-production.api.canada.ca/eng/vehicle-recall-database/v1/recall/make-name/"
+    }
+    url_ <- paste(url_, toString(make), sep = "")
+
+    # api call, returns class vrd_api
+    api_output <- call_vrd_api(url_, make, limit)
+
+    # convert content to a dataframe
+    contents_df <- as.data.frame(api_output$content)
+
+    # currently just provides raw dataframe output, needs to be cleaned up
+    # any repetitive clean-up should go into helper functions
+    # if partial = FALSE need to filter data, if partial = TRUE return all data
+    contents_df
+
+}
+
+#' get_recall_model
+#'
+#' @param model A string
+#' @param limit An integer
+#' @param partial A bool
+#'
+#' @return dataframe
+#'
+#' @examples
+#' \dontrun{
+#' get_recall_model('488')
+#' }
+get_recall_model <- function(model, limit = 25, partial = FALSE) {
+
+    # format the url string
+    url_ <- "https://vrdb-tc-apicast-production.api.canada.ca/eng/vehicle-recall-database/v1/recall/model-name/"
+    url_ <- paste(url_, toString(model), sep = "")
+
+    # api call, returns class vrd_api
+    api_output <- call_vrd_api(url_, model, limit)
+
+    # convert content to a dataframe
+    contents_df <- as.data.frame(api_output$content)
+
+    # currently just provides raw dataframe output, needs to be cleaned up
+    # any repetitive clean-up should go into helper functions
+    # if partial = FALSE need to filter data, if partial = TRUE return all data
+    contents_df
+
+}
+
+#' get_recall_years
+#'
+#' @param start_year An integer
+#' @param end_year An integer
+#' @param limit An integer
+#'
+#' @return dataframe
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' get_recall_years(2010, 2012)
+#' }
+get_recall_years <- function(start_year = 1900, end_year = 2100, limit = 25) {
+
+    # format the url string
+    year_range <- paste(toString(start_year), toString(end_year), sep = "-")
+    url_ <- "https://vrdb-tc-apicast-production.api.canada.ca/eng/vehicle-recall-database/v1/recall/recall-number/"
+    url_ <- paste(url_, year_range, sep = "")
+
+    # api call, returns class vrd_api
+    api_output <- call_vrd_api(url_, year_range, limit)
+
+    # convert content to a dataframe
+    contents_df <- as.data.frame(api_output$content)
+
+    # currently just provides raw dataframe output, needs to be cleaned up
+    # any repetitive clean-up should go into helper functions
+    contents_df
+
+}
+
+#' get_recall_number
+#'
+#' @param recall_number An integer
+#' @param limit An integer
+#'
+#' @return dataframe
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' get_recall_number(1977044)
+#' }
+get_recall_number <- function(recall_number, limit = 25) {
```

Comment:
perhaps
`get_recall_by_number` or `get_recall_from_number` would be better?
`get_recall_number` makes it seem like we are asking for a number, not providing one

Reply:
I'm really disliking these long function names, but I get what you're saying. One could also apply the same logic to the other function names (we're not getting model names from API, were asking for the details for the model names), what about changing to `recall_by_number()` and `recall_by_name()`

Reply:
I don't think 22 characters is a long function name.

But I'm not going to press this, we got bigger fish to fry 🙃